featurebyte/api/view.py

```diff
@@ -1408,6 +1519,54 @@ def _get_input_node_for_lookup_node(self) -> Node:
 
         return node_before_projection
 
+    def get_lookup_node_params(
+        self, column_names: List[str], feature_names: List[str], offset: Optional[str]
+    ) -> Dict[str, Any]:
+        """
+        Get the parameters for the lookup node in as_features().
+
+        Parameters
+        ----------
+        column_names: List[str]
+            List of column names to be used as input to the lookup node
+        feature_names: List[str]
+            List of feature names to be used as output of the lookup node
+        offset: Optional[str]
+            Offset to be used for the lookup node
+
+        Returns
+        -------
+        Dict[str, Any]
+
+        Raises
+        ------
+        ValueError
+            If the entity_column is not found in the columns_info
+        """
+        # Get entity_column
+        entity_column = self.get_join_column()
+
+        # Get serving_name
+        columns_info = self.columns_info
+        column_entity_map = {col.name: col.entity_id for col in columns_info if col.entity_id}
+        if entity_column not in column_entity_map:
+            raise ValueError(f'Column "{entity_column}" is not an entity!')
+        entity_id = column_entity_map[entity_column]
+        entity = Entity.get_by_id(entity_id)
+        serving_name = entity.serving_name
+
+        # Set up Lookup node
+        additional_params = self.get_as_feature_parameters(offset=offset)
```

Comment:
Would it be clearer to rename this to something like `get_additional_lookup_parameters`? The return parameters like `event_parameters` or `scd_parameters` are also relevant for target lookups.

Reply:
updated!